File: src/main/scala/robosky/ether/block/machine/base/BaseMachineBlock.scala
package robosky.ether.block.machine.base

import net.minecraft.block.entity.BlockEntity
import net.minecraft.block.{Block, BlockEntityProvider}
import net.minecraft.util.registry.Registry
import net.minecraft.world.BlockView
import robosky.ether.block.machine.MachineRegistry

abstract class BaseMachineBlock(s: Block.Settings) extends Block(s) with BlockEntityProvider {
  override def createBlockEntity(var1: BlockView): BlockEntity =
    MachineRegistry
      .MACHINES(
        Registry.BLOCK
          .getId(this)
      )
      .instantiate()
}

File: src/main/java/robosky/uplands/block/machine/base/BaseMachineBlock.java
package robosky.uplands.block.machine.base;

import robosky.uplands.block.machine.MachineRegistry;

import net.minecraft.block.Block;
import net.minecraft.block.BlockEntityProvider;
import net.minecraft.block.entity.BlockEntity;
import net.minecraft.util.registry.Registry;
import net.minecraft.world.BlockView;

public abstract class BaseMachineBlock extends Block implements BlockEntityProvider {

    public BaseMachineBlock(Settings settings) {
        super(settings);
    }

    @Override
    public BlockEntity createBlockEntity(BlockView world) {
        return MachineRegistry.MACHINES.get(Registry.BLOCK.getId(this)).instantiate();
    }
}
